src/NUnitFramework/framework/Internal/TestProgressReporter.cs

```diff
@@ -176,6 +181,21 @@ public void TestFinished(ITestResult result)
         #region Helper Methods
 
         /// <summary>
+        /// Returns the parent test item for the targer test item if it exists
+        /// </summary>
+        /// <param name="test"></param>
+        /// <returns>parent test item</returns>
+        private static ITest GetParent(ITest test)
+        {
+            if (test == null || test.Parent == null)
+            {
+                return null;
+            }
+
+            return test.Parent.IsSuite ? test.Parent : GetParent(test.Parent);
```

Comment:
I was commenting on the single line
   return test.Parent.IsSuite ? test.Parent : GetParent(test.Parent);

It should be impossible for test.Parent.IsSuite to be false because only a suite can be a parent.